File: JavaScript/Bit Manipulation/136. Single Number.js
/* 136. Single Number
Difficulty: Easy
Submission Link: https://leetcode.com/problems/single-number/submissions/1386409560/
 */

var singleNumber = function(nums) {
    let result = 0;
    for (let num of nums) {
        result ^= num;
    }
    return result;
};

File: Python/Bit Manipulation/136. Single Number.py
""" 136. Single Number
Difficulty: Easy
Submission Link: https://leetcode.com/problems/single-number/submissions/1386409731/ """

class Solution(object):
    def singleNumber(self, nums):
        x = 0
        for i in range(len(nums)):
            x = x^nums[i]
        return x